File: web/data-structures/HashTable.php
<?php 
require_once 'Entry.php';
require_once 'Hashing.php';

class HashTable {
  public $array;
  public $maxSize;
  public $size;

  function __construct($maxSize) {
    $this->size = 0;
    $this->maxSize = $maxSize;
    $this->array = new SplFixedArray($maxSize);
    for ($i=0; $i < $maxSize; $i++) { 
      $this->array[$i] = null;
    }
  }

  function insert($key, $value) {
    $index = $this->hash($key);
    $this->array[$index] = new Entry($key, $value);
    $this->size++;

    return true;
  }

  function find($key) {
    $index = $this->hash($key);
    $item = $this->array[$index];

    if ($item != null) {
      return $item->value;
    } else {
      return null;
    }
  }

  function delete($key) {
    $index = $this->hash($key);
    $this->array[$index] = null;
    $this->size--;

    return true;
  }

  function hash($key) {
    $value = Hashing::convertKey($key);
    $index = Hashing::multiplyMethod($this->maxSize, $value);

    // Sondeo cuadrático
    $i = 0;
    while ($this->array[$index] != null && $this->array[$index]->key != $key) {
      $i++;
      $index = $index + ($i * $i);
      $index = $index % $this->maxSize;
    }

    return $index;
  }

}



 ?>

File: android/Hash/HashTable.java
package com.example.rolando.myapplication_buenaversiob.Hash;

/**
 * Created by VICTOR MEDINA on 06/12/2016.
 */
public class HashTable<E> {
    Entry<E>[] array;
    int maxSize;
    int size = 0;

    HashTable(int maxSize) {
        this.maxSize = maxSize;
        this.array = new Entry[maxSize];

        for(int i = 0; i < maxSize; ++i) {
            this.array[i] = null;
        }

    }

    public boolean insert(String key, E value) {
        int index = this.hash(key);
        this.array[index] = new Entry(key, value);
        ++this.size;
        return true;
    }

    public E find(String key){
        int index=hash(key);
        Entry<E> item=array[index];
        if(item!=null){
            return item.value;
        }
        return null;
    }

    public boolean delete(String key) {
        int index = this.hash(key);
        this.array[index] = null;
        --this.size;
        return true;
    }

    public int hash(String key) {
        long value = Hashing.convertKey(key);
        int index = Hashing.multiply_method(this.maxSize, value);

        for(int i = 0; this.array[index] != null && !this.array[index].key.equals(key); index %= this.maxSize) {
            ++i;
            index += i * i;
        }

        return index;
    }
}
